File: scala-akka-http-server/src/main/scala/io/swagger/server/model/V3.StopAccessibility.scala
package io.swagger.server.model


/**
 * @param lighting Indicates if there is lighting at the stop
 * @param platform_number Indicates the platform number for xivic information (Platform 0 indicates general stop facilities)
 * @param audio_customer_information Indicates if there is at least one audio customer information at the stop/platform
 * @param escalator Indicates if there is at least one accessible escalator at the stop/platform that complies with the Disability Standards for Accessible Public Transport under the Disability Discrimination Act (1992)
 * @param hearing_loop Indicates if there is a hearing loop facility at the stop/platform
 * @param lift Indicates if there is an elevator at the stop/platform
 * @param stairs Indicates if there are stairs available in the stop
 * @param stop_accessible Indicates if the stop is accessible
 * @param tactile_ground_surface_indicator Indicates if there are tactile tiles (also known as tactile ground surface indicators, or TGSIs) at the stop
 * @param waiting_room Indicates if there is a general waiting area at the stop
 * @param wheelchair 
 */
case class V3.StopAccessibility (
  lighting: Option[Boolean],
  platform_number: Option[Int],
  audio_customer_information: Option[Boolean],
  escalator: Option[Boolean],
  hearing_loop: Option[Boolean],
  lift: Option[Boolean],
  stairs: Option[Boolean],
  stop_accessible: Option[Boolean],
  tactile_ground_surface_indicator: Option[Boolean],
  waiting_room: Option[Boolean],
  wheelchair: Option[V3.StopAccessibilityWheelchair]
)

File: dart/lib/model/v3_stop_accessibility.dart
part of swagger.api;

class V3StopAccessibility {
  /* Indicates if there is lighting at the stop */
  bool lighting = null;
/* Indicates the platform number for xivic information (Platform 0 indicates general stop facilities) */
  int platformNumber = null;
/* Indicates if there is at least one audio customer information at the stop/platform */
  bool audioCustomerInformation = null;
/* Indicates if there is at least one accessible escalator at the stop/platform that complies with the Disability Standards for Accessible Public Transport under the Disability Discrimination Act (1992) */
  bool escalator = null;
/* Indicates if there is a hearing loop facility at the stop/platform */
  bool hearingLoop = null;
/* Indicates if there is an elevator at the stop/platform */
  bool lift = null;
/* Indicates if there are stairs available in the stop */
  bool stairs = null;
/* Indicates if the stop is accessible */
  bool stopAccessible = null;
/* Indicates if there are tactile tiles (also known as tactile ground surface indicators, or TGSIs) at the stop */
  bool tactileGroundSurfaceIndicator = null;
/* Indicates if there is a general waiting area at the stop */
  bool waitingRoom = null;

  V3StopAccessibilityWheelchair wheelchair = null;

  V3StopAccessibility();

  @override
  String toString() {
    return 'V3StopAccessibility[lighting=$lighting, platformNumber=$platformNumber, audioCustomerInformation=$audioCustomerInformation, escalator=$escalator, hearingLoop=$hearingLoop, lift=$lift, stairs=$stairs, stopAccessible=$stopAccessible, tactileGroundSurfaceIndicator=$tactileGroundSurfaceIndicator, waitingRoom=$waitingRoom, wheelchair=$wheelchair, ]';
  }

  V3StopAccessibility.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    lighting = json['lighting'];
    platformNumber = json['platform_number'];
    audioCustomerInformation = json['audio_customer_information'];
    escalator = json['escalator'];
    hearingLoop = json['hearing_loop'];
    lift = json['lift'];
    stairs = json['stairs'];
    stopAccessible = json['stop_accessible'];
    tactileGroundSurfaceIndicator = json['tactile_ground_surface_indicator'];
    waitingRoom = json['waiting_room'];
    wheelchair = new V3StopAccessibilityWheelchair.fromJson(json['wheelchair']);
  }

  Map<String, dynamic> toJson() {
    return {
      'lighting': lighting,
      'platform_number': platformNumber,
      'audio_customer_information': audioCustomerInformation,
      'escalator': escalator,
      'hearing_loop': hearingLoop,
      'lift': lift,
      'stairs': stairs,
      'stop_accessible': stopAccessible,
      'tactile_ground_surface_indicator': tactileGroundSurfaceIndicator,
      'waiting_room': waitingRoom,
      'wheelchair': wheelchair
     };
  }

  static List<V3StopAccessibility> listFromJson(List<dynamic> json) {
    return json == null ? new List<V3StopAccessibility>() : json.map((value) => new V3StopAccessibility.fromJson(value)).toList();
  }

  static Map<String, V3StopAccessibility> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, V3StopAccessibility>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new V3StopAccessibility.fromJson(value));
    }
    return map;
  }
}
